### PR TITLE
Add github action to upload cloudformation template, fix template

### LINKED ISSUE #20: Upload latest deployment/aws/cloudformation/ecs/posthog.yaml to S3
Currently, install instructions point to old file, which runs into problems with e.g. CacheNodeType.

Ideally we could also automate this?

### PR DESCRIPTION
**Pull request by macobo**
Successful run: https://github.com/PostHog/deployment/runs/1258572632?check_suite_focus=true

Closes #20

### CODE REVIEW

File: .github/workflows/upload-to-s3.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    # branches to consider in the event; optional, defaults to all
+    branches:
+      - master
+
+jobs:
+  update_cloudformation_template:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: jakejarvis/s3-sync-action@v0.5.1
```
Comment (fuziontech) [inline .github/workflows/upload-to-s3.yml:14]: ROFL amazing that this repo and container exists 💯

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "cfnLint.path": "/home/macobo/.pyenv/shims/cfn-lint"
+}
```
Comment (fuziontech) [inline .vscode/settings.json]: might want to add this to `.gitignore`

File: aws/cloudformation/ecs/posthog.yaml [hunk 3]
```diff
@@ -618,22 +618,22 @@ Resources:
               Value: !Ref 'SmtpUseSSL'
             - Name: DEFAULT_FROM_EMAIL
               Value: !Ref 'DefaultFromEmail'
-        - Name: !Join [ !Ref 'AWS::StackName', '-worker-beat' ]
+        - Name: !Join [ '', [ !Ref 'AWS::StackName', '-worker-beat' ] ]
```
Comment (fuziontech) [inline aws/cloudformation/ecs/posthog.yaml:621]: Nice catch!

**Review by fuziontech — APPROVED** [review]
lgtm! I'd just add your `.vscode` configs to the `.gitignore` since they look pretty specific to your laptop

File: .vscode/settings.json (hunk 2 above)
Comment (macobo) [reply .vscode/settings.json]: Oops. Didn't intend to commit this. Fixed :facepalm:

File: .github/workflows/upload-to-s3.yml (hunk 1 above)
Comment (macobo) [reply .github/workflows/upload-to-s3.yml:14]: /shrug There were actually several, one of which didn't work.

**Pull request merged**